### ISSUE TITLE
`directoryPath` allway emtpy and throw exception.

### ISSUE THREAD

**Issue by xljiulang** [dotnet/ClangSharp#507]
```
_outputStreamFactory = outputStreamFactory ?? ((path) => {
                var directoryPath = Path.GetDirectoryName(path) ?? "";
                _ = Directory.CreateDirectory(directoryPath);
                return new FileStream(path, FileMode.Create);
            });
```
https://github.com/dotnet/ClangSharp/blob/03746e376f40a5180af84f4981fa5a3c8ac654e7/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs#L135C55-L135C55

**Comment by kkukshtel** [comment]
Came here to post the same — this specifically seems like an issue when using `single-file` instead of `multi-file`. Running the same rsp's with multi-file doesn't generate the same error.

```
Unhandled exception: System.ArgumentException: The value cannot be an empty string. (Parameter 'path')
   at System.ArgumentException.ThrowNullOrEmptyException(String argument, String paramName)
   at System.IO.Directory.CreateDirectory(String path)
   at ClangSharp.PInvokeGenerator.<>c.<.ctor>b__44_0(String path) in /_/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs:line 133
   at ClangSharp.PInvokeGenerator.Close() in /_/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs:line 243
   at ClangSharp.PInvokeGenerator.Dispose(Boolean isDisposing) in /_/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs:line 2227
   at ClangSharp.PInvokeGenerator.Dispose() in /_/sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs:line 1569
   at ClangSharp.Program.Run(InvocationContext context) in /_/sources/ClangSharpPInvokeGenerator/Program.cs:line 800
   at System.CommandLine.Handler.<>c__DisplayClass1_0`1.<SetHandler>b__0(InvocationContext context)
   at System.CommandLine.Invocation.AnonymousCommandHandler.<>c__DisplayClass2_0.<.ctor>g__Handle|0(InvocationContext context)
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass18_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass13_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass20_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__19_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__6_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass9_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

**Comment by kkukshtel** [comment]
Just to bump this @tannergooding - this fully prevents output from working right now when single-file is selected.

**Comment by tannergooding** [comment]
This should already be fixed in local builds, a patch release just hasn’t been published yet.

https://github.com/dotnet/ClangSharp/pull/506

**Comment by tannergooding** [comment]
This was published with v17.0.1
